the-collab-lab/tcl-26-smart-shopping-list
# Show/hide all item details

## Description

Picking up from Connie's PR, I made a new copy that doesn't have all the extra commits and unrelated files.

This adds the "Show/hide all item details" button link to the top of the Shopping List in ListHeader.

![image](https://user-images.githubusercontent.com/61371242/131603192-b967917b-60aa-413b-87e0-342f4de7eae4.png)

A new `showAllDetails` state is toggled on and off by the 'Show/hide all item details' link in the header. This showAllDetails value is passed down into each ShoppingListItem component. A useEffect in the ShoppingListItem component checks for any changes to showAllDetails and toggles the individual items' states accordingly.

Note: Connie changed the `showDetails` state variable in each ShoppingListItem to `showSingleDetail` for clarity.

## Related Issue

Closes #82 : Add "Show all details" link

## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
| ✓ | :sparkles: New feature     |
|    | :hammer: Refactoring       |
|    | :100: Add tests            |
|    | :link: Update dependencies |
|    | :scroll: Docs              |

## Updates

### Before



![image](https://user-images.githubusercontent.com/61371242/131603387-d5561b76-f765-4046-9511-519a2136f78e.png)

### After




## Testing Steps / QA Criteria



1. Pull branch, run npm start.
2. Click "Show all item details" and all item accordions expand and the link should become "Hide all item details."
3. Click "Hide all item details" and all item accordions should close.
4. Individual items should open and close unrelated to the state of the "Show/hide all item details" link, but will be overridden once that link is clicked.
